transformers/file_utils.py

```diff
@@ -334,59 +335,60 @@ def get_from_cache(url, cache_dir=None, force_download=False, proxies=None, etag
     # If we don't have a connection (etag is None) and can't identify the file
     # try to get the last downloaded one
     if not os.path.exists(cache_path) and etag is None:
-        matching_files = fnmatch.filter(os.listdir(cache_dir), filename + '.*')
-        matching_files = list(filter(lambda s: not s.endswith('.json'), matching_files))
+        matching_files = [
+            file
+            for file in fnmatch.filter(os.listdir(cache_dir), filename + '.*')
+            if not file.endswith('.json') and not file.endswith('.lock')
+        ]
         if matching_files:
             cache_path = os.path.join(cache_dir, matching_files[-1])
 
-    if resume_download:
-        incomplete_path = cache_path + '.incomplete'
-        @contextmanager
-        def _resumable_file_manager():
-            with open(incomplete_path,'a+b') as f:
-                yield f
-            os.remove(incomplete_path)
-        temp_file_manager = _resumable_file_manager
-        if os.path.exists(incomplete_path):
-            resume_size = os.stat(incomplete_path).st_size
+    # Prevent parallel downloads of the same file with a lock.
```

Comment:
Is it locking safe even in the case of parallel (and independent) python processes?

If so, we can now update as well the examples in `./examples/` which uses a complex (and PyTorch specific) logic based on `torch.distributed.barrier` (https://pytorch.org/docs/stable/distributed.html#torch.distributed.barrier) to avoid parallel downloading and access to downloaded files.

Reply:
Yes, it's a filesystem lock, so it's safe in the presence of multiple processes.
Also, it disappears automatically if a process dies before releasing the lock.
Indeed, with only one machine, a simple filesystem lock should do the job.